EIPS/eip-3541.md

```diff
@@ -0,0 +1,50 @@
+---
+eip: 3541
+title: Reject new contracts starting with the 0xEF byte
+author: Alex Beregszaszi (@axic), Paweł Bylica (@chfast), Andrei Maiboroda (@gumb0)
+discussions-to: https://ethereum-magicians.org/t/evm-object-format-eof/5727/4
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-03-16
+---
+
+## Abstract
+
+Disallow new code starting with the `0xEF` byte to be deployed. Code already existing in the account trie starting with `0xEF` byte is not affected semantically by this change.
+
+## Motivation
+
+Contracts conforming to the EVM Object Format (EOF), described by [EIP-3540](./eip-3540.md), are going to be validated at deploy time. In order to guarantee that every EOF-formatted contract in the state is valid, we need to prevent already deployed (and not validated) contracts from being recognized as such format. This will be achieved by choosing a byte sequence for the *magic* that doesn't exist in any of the already deployed contracts. To prevent the growth of the search space and to limit the analysis to the contracts existing before this fork, we disallow the starting byte of the format (the first byte of the magic).
+
+## Specification
+
+After `block.number == HF_BLOCK` new contract creation (via create transaction, `CREATE` or `CREATE2` instructions) results in an exceptional abort if the _code_'s first byte is `0xEF`. 
```

Comment:
```suggestion
After `block.number == HF_BLOCK` the code deployment via create transaction, `CREATE` or `CREATE2` instructions results in an exceptional abort if the _deployed code_'s first byte is `0xEF`. 
```

This clarifies that the abort happens after the init code has been executed.

Reply:
This hits the naming issue we are currently struggle with. Yellow Paper defined _initcode_ and _code_ where the later means the deployed code. This is very confusing already in this case and only gets worse when EOF1 introduces code section.

I was carefully avoiding "deploy" name in the spec because this is not properly defined. But maybe we should introduce more precise names instead.